src/sdk/namespace_fs.js

```diff
@@ -2021,6 +2021,12 @@ class NamespaceFS {
         dbg.log0('namespace_fs.restore_object:', params);
 
         await this._throw_if_storage_class_not_supported(s3_utils.STORAGE_CLASS_GLACIER);
+        if (params.days > config.NSFS_GLACIER_RESTORE_MAX_DAYS) {
+            throw new S3Error({
+                ...S3Error.BadRequest,
+                detail: `restore days cannot be more than ${config.NSFS_GLACIER_RESTORE_MAX_DAYS}`},
+            );
+        }
```

Comment:
I am not sure about failing. Perhaps we should just truncate the number?

Reply:
Yeah, I think adding another option for the behaviour is better.